Automattic/jetpack
# Tiled gallery: remove Photon module dependency

Make Tiled gallery block available without Photon module being active.

Requiring users to turn on "image optimizations" is confusing and arbitrary, and we don't depend on that technically either.

If there's an agency/partner that wants to hide the block, they shouldn't need to disable photon for doing that and there's a separate mechanism in Jetpack for hiding specific blocks.

It's not great to turn on Photon module for the whole page load either (that's what current tiled gallery does). All the other images on the page should be un-affected by Tiled gallery's dependency on Photon servers.

Resolves https://github.com/Automattic/jetpack/issues/13470

Similar question for Markdown block: https://github.com/Automattic/jetpack/pull/13473

#### Changes proposed in this Pull Request:
- Remove Photon class & module check in block registration

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
Changing

#### Testing instructions:
- Have a Jetpack site (e.g. via https://jurassic.ninja/)
- Confirm that Tiled gallery block is available in block editor only when Photon module is active. You can activate/disable it via "Performance -> Speed up image load times" toggle in Jetpack settings:
- In this branch (Test e.g. via https://jurassic.ninja/create?jetpack-beta&branch=update/tiled-gallery-photon-dep&wp-debug-log)

![image](https://user-images.githubusercontent.com/87168/64922999-7b501e00-d7a3-11e9-9cba-74c2326301a6.png)

![image](https://user-images.githubusercontent.com/87168/64923028-ec8fd100-d7a3-11e9-8f7a-353fb6e80a8b.png)



#### Proposed changelog entry for your changes:
- Tiled gallery block is now available in block picker even if "image accelerator" feature is disabled.
